test/integration/good/compiler-optimizations/cpp.expected

```diff
@@ -28138,6 +28196,475 @@ stan::math::profile_map& get_stan_profile_data() {
+ " (in 'unenforce-initialize.stan', line 10, column 2 to line 12, column 3)",
+ " (in 'unenforce-initialize.stan', line 15, column 4 to column 21)",
+ " (in 'unenforce-initialize.stan', line 14, column 2 to line 16, column 3)"};
+
+
+
+class unenforce_initialize_model final : public model_base_crtp<unenforce_initialize_model> {
+
+ private:
+   
+  
+ 
+ public:
+  ~unenforce_initialize_model() { }
+  
+  inline std::string model_name() const final { return "unenforce_initialize_model"; }
+
+  inline std::vector<std::string> model_compile_info() const noexcept {
+    return std::vector<std::string>{"stanc_version = %%NAME%%3 %%VERSION%%", "stancflags = --O --print-cpp"};
+  }
+  
+  
+  unenforce_initialize_model(stan::io::var_context& context__,
+                             unsigned int random_seed__ = 0,
+                             std::ostream* pstream__ = nullptr) : model_base_crtp(0) {
+    int current_statement__ = 0;
+    using local_scalar_t__ = double ;
+    boost::ecuyer1988 base_rng__ = 
+        stan::services::util::create_rng(random_seed__, 0);
+    (void) base_rng__;  // suppress unused var warning
+    static constexpr const char* function__ = "unenforce_initialize_model_namespace::unenforce_initialize_model";
+    (void) function__;  // suppress unused var warning
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+    try {
+      int pos__;
+      pos__ = 1;
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+    }
+    num_params_r__ = 1;
+    
+  }
+  
+  template <bool propto__, bool jacobian__ , typename VecR, typename VecI, 
+  stan::require_vector_like_t<VecR>* = nullptr, 
+  stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr> 
+  inline stan::scalar_type_t<VecR> log_prob_impl(VecR& params_r__,
+                                                 VecI& params_i__,
+                                                 std::ostream* pstream__ = nullptr) const {
+    using T__ = stan::scalar_type_t<VecR>;
+    using local_scalar_t__ = T__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    stan::io::deserializer<local_scalar_t__> in__(params_r__, params_i__);
+    int current_statement__ = 0;
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+    static constexpr const char* function__ = "unenforce_initialize_model_namespace::log_prob";
+    (void) function__;  // suppress unused var warning
+    
+    try {
+      double lcm_sym6__;
+      double lcm_sym5__;
+      local_scalar_t__ y;
+      current_statement__ = 1;
+      y = in__.template read<local_scalar_t__>();
+      double no_init = std::numeric_limits<double>::quiet_NaN();
+      local_scalar_t__ init_from_param;
+      current_statement__ = 3;
+      init_from_param = (y * 5.0);
```

Comment:
I'm confused why `init_from_param` isn't initialized here? In `O0` (this file) it should still produce a std::numeric_limits call, right?

Reply:
Oh I understand, `cpp.expected` is `-Oexperimental`. I figured it would be `O0` - which we should add as another test output for comparison.

Reply:
^Yeah that would be nice to do but lets do it in a separate PR so this one doesn't get huge